delta/kernel/src/main/scala/ch/epfl/bluebrain/nexus/delta/kernel/jwt/ParsedToken.scala

```diff
@@ -18,7 +25,24 @@ final case class ParsedToken private (
     expirationTime: Instant,
     groups: Option[Set[String]],
     jwtToken: SignedJWT
-)
+) {
+
+  def validate(audiences: Option[NonEmptySet[String]], keySet: JWKSet): Either[InvalidAccessToken, Unit] = {
+    val proc        = new DefaultJWTProcessor[SecurityContext]
+    val keySelector = new JWSVerificationKeySelector(JWSAlgorithm.RS256, new ImmutableJWKSet[SecurityContext](keySet))
+    proc.setJWSKeySelector(keySelector)
+    audiences.foreach { aud =>
+      proc.setJWTClaimsSetVerifier(new DefaultJWTClaimsVerifier(aud.toSet.asJava, null, null, null))
+    }
+    Either
+      .catchNonFatal(proc.process(jwtToken, null))
+      .bimap(
+        err => InvalidAccessToken(subject, issuer, err.getMessage),
+        _ => ()
+      )
+  }
```

Comment:
As someone unfamiliar with this process, it's not clear what an audience or keySet is. And it's not clear what about them is being validated

I think with some name changes and perhaps pulling some code into private methods this could be made really clear.

Reply:
This part is already documented in the realm documentation:
https://bluebrainnexus.io/snapshot/docs/delta/api/realms-api.html#payload
And in the configuration file